### PR TITLE
COMPASS-3613 Add name to connection and save connection to favourites

### PR DESCRIPTION
**Pull request by alenakhineika**
## Description

- Add the favorite pill
- Show modal when clicking on the favorite pill
- Add/edit favorite name
- Save/delete favorite

### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/components/form/favorite-modal.jsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  }
+
+  /**
+   * Submits the form.
+   *
+   * @param {Object} evt - evt.
+   */
+  handleFormSubmit(evt) {
+    evt.preventDefault();
+    evt.stopPropagation();
```
Comment (lrlna) [inline src/components/form/favorite-modal.jsx]: to double check: you want to do both of these preventions, yea?

**Comment by sampoonachot** [comment]
Re: @lrlna's comment for #2 

If i recall from durran, as of right now, we're unable to parse the information from the form into the string (tho we can go from string to form).

For #3 on alignment, can we move the link (paste connection string) to a row below the connection name and pill? it looks like they're part of the row/container.

**Comment by lrlna** [comment]
@sampoonachot sorry i was not very clear about 2. What I meant is that the favourite name does not carry forward, e.g. in this gif (also posted it in slack): 
IVlqThQqkK

File: src/components/form/favorite-modal.jsx (hunk 1 above)
Comment (alenakhineika) [reply src/components/form/favorite-modal.jsx]: yes! :)

**Comment by alenakhineika** [comment]
@lrlna @sampoonachot updated!

**Comment by lrlna** [comment]
hmmmm there is still something off with the way that top heading works. Both the `name` of the connection, the `favourite` button, and the text on the other side drop down when the name is too long. Maybe we should use an ellipsis after a certain number of characters? 
Screen Shot 2019-09-30 at 11 29 39 AM
Screen Shot 2019-09-30 at 11 29 30 AM

**Comment by alenakhineika** [comment]
@lrlna i cut the string after certain number of characters. will take a look why it does not work for you. for me it works even for long strings.

**Comment by lrlna** [comment]
sorry my git didn't want to fetch the latest commit so I had some stale
commits on my branch. It looks good!


Le lun. 30 sept. 2019 à 11:34, Alena Khineika <notifications@github.com> a
écrit :

**Pull request merged**